test/credentials.spec.js

```diff
@@ -113,12 +113,54 @@
     });
   });
 
+  describe('refreshable credentials', function() {
+    AWS.util.arrayEach(
+      [
+        'CognitoIdentityCredentials',
+        'ChainableTemporaryCredentials',
+        'TemporaryCredentials',
+        'EC2MetadataCredentials',
+        'ECSCredentials',
+        'RemoteCredentials',
+        'SAMLCredentials',
+        'SharedIniFileCredentials',
+        'WebIdentityCredentials'
+      ],
+      function(credClass) {
+        if (credClass in AWS) {
+          it(credClass + ' coalesces concurrent calls',
+            function (done) {
+              var callCount = 10;
+              var callbackCount = 0;
+              var creds = new AWS[credClass]({});
+              var loadSpy = helpers.spyOn(creds, 'load').andCallFake(
+                function(callback) {
+                  setImmediate(function () { callback(null); });
```

Comment:
`setImmediate` is not a standard API. It may break in some versions of headless Chrome. You can use `setTimeout`. I don't know why it passes the CI but this test case should only run in node environment because some providers(like `EC2MetadataCredentials`) are only available in Node(see `AWS.util.isNode()`)

Reply:
The available providers are filtered by the `if` wrapping the test case, so the tests only run for classes which are properties of `AWS`. `setImmediate` is poly-filled by browserify, but I can swap it for `AWS.util.defer` (which will end up using the polly-fill) - I didn't use `setTimeout` because it is patched with a synchronous function when the tests run.